front-end/src/components/layout/Breadcrumbs.js

```diff
@@ -0,0 +1,57 @@
+import { useLocation } from "react-router-dom";
+import MUIBreadcrumbs from "@mui/material/Breadcrumbs";
+import Paper from "@mui/material/Paper";
+import Container from "@mui/system/Container";
+
+import Link from "./Link";
+
+import "./Breadcrumbs.css";
+
+function getPathFriendlyName(path) {
+  const friendlyNames = {
+    register: "Register",
+    login: "Login",
+    find: "Find Tables",
+    "view-table/d+": "View Table",
+    "submit-table": "Submit Table",
+  };
+
+  const entries = Object.entries(friendlyNames);
+  const lookup = entries.map(([regex, name]) => ({ regex, name }));
+  const entry = lookup.find(({ regex }) => path.match(regex));
+
+  return entry?.name;
+}
+
+function createPathBreadcrumbLink(_, index, paths) {
+  const urlPath = paths.slice(0, index + 1).join("/");
+  const friendlyName = getPathFriendlyName(urlPath);
+
+  if (!friendlyName) {
+    return undefined;
+  }
+
+  return (
+    <Link underline="hover" to={urlPath} key={urlPath}>
+      {friendlyName}
+    </Link>
+  );
+}
+
+export default function Breadcrumbs() {
+  const location = useLocation();
+  const paths = location.pathname.split("/").filter((x) => !!x);

```

Comment:
With the current way of creating friendly names, we would need to nest the routes. I think we could also *not* nest the routes and basically add a special case for View Table that inserts the find table breadcrumb.

One other thing I'll note is that because the View Table screen displays in a dialog, the breadcrumbs don't appear on that screen. I'm not 100% sure why, my thought is that the Dialog component doesn't use the same routing logic or something?

Or I've just misunderstood how the router works.

Reply:
I'm up for nesting the routes, or not doing it... maybe because there are multiple ways to get to the View Table page, we could make it a custom case for now.

I'm not sure exactly what's happening with that... I think the `ViewTable` shouldn't be in a dialog, but the `TablePopup` is - is that what you meant? I think it's fine to not show the breadcrumbs in the dialog. But they were showing up on the `ViewTable` for me.

Reply:
Oh, maybe I'm getting `TablePopup` confused with `ViewTable`, then. I was talking about whichever view displayed in a dialog.

Reply:
Tracked in #83